tests/backup/backup_test.go

```diff
@@ -237,6 +237,157 @@ var _ = Describe("{CreateMultipleUsersAndGroups}", func() {
+				cloudCredUID = uuid.New()
+				CloudCredUidList = append(CloudCredUidList, cloudCredUID)
+				backupLocationUID = uuid.New()
+				credName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				CreateCloudCredential(provider, credName, cloudCredUID, orgID)
+				log.InfoD("Created Cloud Credentials with name - %s", credName)
+				time.Sleep(time.Minute * 1)
+				BackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				CreateBackupLocation(provider, BackupLocationName, backupLocationUID, credName, cloudCredUID, bucketName, orgID, "")
+				log.InfoD("Created Backup Location with name - %s", BackupLocationName)
+			}
+		})
+		Step("Register source and destination cluster for backup", func() {
+			log.InfoD("Registering Source and Destination clusters and verifying the status")
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster status")
+		})
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking Backup of application")
+			err = CreateBackup(backupName, SourceClusterName, BackupLocationName, backupLocationUID, []string{bkpNamespaces[0]},
+				nil, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, "Verifying backup creation")
+		})
+
+		Step("Share backup with user", func() {
+			log.InfoD("Share backup with  user having full access")
+			err := ShareBackup(backupName, nil, []string{userName}, FullAccess, ctx)
+			log.FailOnError(err, "Failed to share backup %s", backupName)
+		})
+
+		Step("Duplicate shared backup", func() {
+			log.InfoD("Validating to duplicate share backup without adding clutser")
+			// Get user context
+			ctxNonAdmin, err := backup.GetNonAdminCtx(userName, password)
+			log.FailOnError(err, "Fetching non px-central-admin user ctx")
+			userContexts = append(userContexts, ctxNonAdmin)
+
+			// Validate that backups are shared with user
+			log.Infof("Validating that backups are shared with %s user", userName)
+			userBackups1, _ := GetAllBackupsForUser(userName, password)
+			dash.VerifyFatal(len(userBackups1), numberOfBackups, fmt.Sprintf("Validating that user [%s] has access to all shared backups", userName))
+
+			//to duplicate shared backup internally it calls create backup api
+			log.Infof("Duplicate shared backup")
+			err = CreateBackup(backupName, SourceClusterName, BackupLocationName, backupLocationUID, []string{bkpNamespaces[0]},
+				nil, orgID, clusterUid, "", "", "", "", ctxNonAdmin)
+			log.Infof("user not able to duplicate shared backup without adding cluster with err - %v", err)
+			dash.VerifyFatal(strings.Contains(err.Error(), "NotFound desc = failed to retrieve cluster [source-cluster]: object not found"), true, "Verifying that shared backup can't be duplicated without adding cluster")
```

Comment:
Here the string `source-cluster` is coming form the variable `SourceClusterName`.
Please replace `source-cluster` with `%s` and assign it to whatever is there at `SourceClusterName`

Reply:
Added SourceClusterName string in dash.VerifyFatal for reference